layer_test.go

```diff
@@ -24,3 +24,26 @@ func TestCreateLayerNoSources(t *testing.T) {
 	_, err := CreateLayer(config)
 	assert.Equal(t, NoSourcesErr, err, "Expected to fail due to no sources for layer")
 }
+
+func TestCreateLayerMinZoomOutOfBounds(t *testing.T) {
+	config := LayerConfig{
+		Minzoom: MinZoom - 1,
+		Maxzoom: MaxZoom - 1,
+		Source: SourceConfig{
+			Elasticsearch: new(ElasticsearchConfig),
+		},
+	}
+	_, err := CreateLayer(config)
+	assert.Equal(t, LayerMinZoomOutOfBoundsErr, err, "Expected to fail because layer min zoom is less than absolute allowed min")
+}
+func TestCreateLayerMaxZoomOutOfBounds(t *testing.T) {
```

Comment:
As it stands, a `TestCreateLayerInBounds` test, where all layerConfig params are valid, will still fail the unit test because `CreateLayer` attempts a call out to the backend server if the config is valid.
A possible way to clean this up would be to add validate methods to any go struct that unmarshals a yaml config.
```golang
func (s *layerConfig) Validate() error {...}
```
that would remove the validation layer out of the logic of `CreateLayer`, and allow it to be called on its own. separating this out into its own function would allow for successful tests in layer_test.go by calling `config.Validate()` instead of `CreateLayer(config)`

Reply:
That seems like a good plan to me @brian-dellabetta. IMO we should get your recommended change in before merge, not only to fix this unit test you mentioned, but also since this will make further pre-boot sanity checks simpler to integrate in the future!